src/abstractdataframe/selectionfast.jl

```diff
@@ -0,0 +1,243 @@
+    end
+    return len
+end
+
+function table_transformation(df_sel::AbstractDataFrame, ::typeof(mean))
+    @assert ncol(df_sel) > 0
+    return mean(map(identity, eachcol(df_sel)))
+end
+
+table_transformation(df_sel::AbstractDataFrame, ::ByRow{typeof(mean)}) =
+    table_transformation(df_sel, mean)
+
+function table_transformation(df_sel::AbstractDataFrame,
+                                        fun::typeof(ByRow(mean∘skipmissing)))
+    fastmean = _mean_skipmissing_fast(map(identity, eachcol(df_sel)))
+    isnothing(fastmean) || return fastmean
+    slowmean = default_table_transformation(df_sel, fun)
+    isconcretetype(nonmissingtype(eltype(slowmean))) && return slowmean
+    T = mapreduce(typeof, promote_type, slowmean)
+    return convert(AbstractVector{T}, slowmean)
+end
+
+function _mean_skipmissing_fast(cols::Vector{<:AbstractVector})
+    local sumz
+    sumz_undefined = true
+    for col in cols
+        T = nonmissingtype(eltype(col))
+        T === Union{} && continue
+        # if T is not concrete we cannot reliably implement fast path
+        isconcretetype(T) || return nothing
+        try
+            zi = zero(T) / 1
+            if sumz_undefined
+                sumz_undefined = false
+                sumz = zi
+            else
+                sumz += zi
+            end
+        catch e
+            if e isa MethodError && e.f === zero
+                sumz_undefined = true
+                break
+            else
+                throw(e)
+            end
+        end
+    end
+    # this will happen if eltype of some columns do not support zero
+    # or all columns have eltype Missing
+    sumz_undefined && return nothing
+    sumv = fill!(Tables.allocatecolumn(typeof(sumz), length(cols[1])), sumz)
+    lenv = zeros(Int, length(sumv))
+
+    for col in cols
+        sumv .= ifelse.(ismissing.(col), sumv, sumv .+ col)
+        lenv .+= .!ismissing.(col)
+    end
+    sumv ./= lenv
+    return sumv
+end
+
+table_transformation(df_sel::AbstractDataFrame, ::typeof(ByRow(minimum))) =
+    _minmax_row_fast(map(identity, eachcol(df_sel)), min)
+
+table_transformation(df_sel::AbstractDataFrame, ::typeof(ByRow(maximum))) =
+    _minmax_row_fast(map(identity, eachcol(df_sel)), max)
+
+function _minmax_row_fast(cols::Vector{<:AbstractVector},
+                          fun::Union{typeof(min), typeof(max)})
+    isempty(cols) && throw(ArgumentError("No columns selected for reduction"))
+    T = mapreduce(typeof, promote_type, cols)
+    res = Tables.allocatecolumn(T, length(cols[1]))
+    res .= cols[1]
+    for i in 2:length(cols)
+        res .= fun.(res, cols[i])
+    end
+    return res
+end
+
+function table_transformation(df_sel::AbstractDataFrame, ::typeof(minimum))
+    return reduce(min, map(identity, eachcol(df_sel)))
+end
+
+function table_transformation(df_sel::AbstractDataFrame, ::typeof(maximum))
+    return reduce(max, map(identity, eachcol(df_sel)))
+end
+
+# TODO:
```

Comment:
@nalimilan - I have made a list of additional standard reductions that might be asked for. We can add all of them (the list is long, or maybe we split this PR into two and concentrate only on some of them here and I would add other in the follow up - this maybe would be easier to review?)